cmd/kubeadm/app/util/endpoint.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+// GenerateMasterEndpoint generates a properly formated Master URL.
+func GenerateMasterEndpoint(protocol, address string, port int32) string {
```

Comment:
Wondering if this should use the Go idiom of returning a value and an error object, rather that having the caller check the value for empty string an then create an error. Also allows you to differentiate between invalid protocol and invalid address.